java/server/src/org/openqa/grid/web/servlet/HubStatusServlet.java

```diff
@@ -136,16 +136,14 @@ private JsonObject getResponse(HttpServletRequest request) throws IOException {
   private JsonObject getSlotCounts() {
     int freeSlots = 0;
     int totalSlots = 0;
+    int usedSlots = 0;
 
     for (RemoteProxy proxy : getRegistry().getAllProxies()) {
-      for (TestSlot slot : proxy.getTestSlots()) {
-        if (slot.getSession() == null) {
-          freeSlots += 1;
-        }
-
-        totalSlots += 1;
-      }
+      totalSlots += proxy.getTestSlots().size() > proxy.getMaxNumberOfConcurrentTestSessions() ?
```

Comment:
reducing this further... could just get rid of the two other variables and just have:

    freeSlots += Math.min(proxy.getMaxNumberOfConcurrentTestSessions(), proxy.getTestSlots().size()) - proxy.getTotalUsed();

and yeah, i agree still need the `+=` :)